MSAL/src/public/MSALError.h

```diff
@@ -117,235 +115,223 @@ extern NSString *MSALBrokerVersionKey;
  */
 extern NSString *MSALHomeAccountIdKey;
 
-typedef NS_ENUM(NSInteger, MSALErrorCode)
-{
-    /*!
-        A required parameter was not provided, or a passed in parameter was
-        invalid. See MSALErrorDescriptionKey for more information.
-     */
-    MSALErrorInvalidParameter = -42000,
-    
-    /*!
-        The required MSAL URL scheme is not registered in the app's info.plist.
-        The scheme should be "msal<clientid>"
-     
-        e.g. an app with the client ID "abcde-12345-vwxyz-67890" would need to
-        register the scheme "msalabcde-12345-vwxyz-67890" and add the
-        following to the info.plist file:
-     
-        <key>CFBundleURLTypes</key>
-        <array>
-            <dict>
-                <key>CFBundleTypeRole</key>
-                <string>Editor</string>
-                <key>CFBundleURLName</key>
-                <string>$(PRODUCT_BUNDLE_IDENTIFIER)</string>
-                <key>CFBundleURLSchemes</key>
-                <array>
-                    <string>msalabcde-12345-vwxyz-67890</string>
-                </array>
-            </dict>
-
-     */
-    MSALErrorRedirectSchemeNotRegistered    = -42001,
+extern NSString *MSALErrorDomain;
 
-    MSALErrorInvalidRequest                 = -42002,
-    MSALErrorInvalidClient                  = -42003,
-    MSALErrorInvalidGrant                   = -42004,
-    MSALErrorInvalidScope                   = -42005,
-    MSALErrorUnauthorizedClient             = -42006,
+// TODO: update error numbers for each enum?
+typedef NS_ENUM(NSInteger, MSALServiceErrorCode)
```

Comment:
I like categorization like this. I do see one problem with this approach, vs generic error code approach. Because it's enum, there's no easy way to check if the error I received is from this particular error category (like in Android case they can check exception type). I see two options to improve:

1. Provide a generic error code, e.g. MSALServiceError and additional error info is somewhere additional
2. Provide helper functions to detect which error category error belongs to, e.g. isServiceError(NSInteger serviceErrorCode), or isRecoverableError etc. 

I also see a bit of a challenge that we now have different enums. Let's maybe check tomorrow if there any example in Apple error codes regarding categorizing error codes and how they solved it.

Reply:
Another option could be introducing subclasses of NSError, like MSALServiceError, MSALClientError etc.